provider/fs/src/fs_data_provider.rs

```diff
@@ -79,35 +80,49 @@ impl FsDataProvider {
         };
         Ok((BufReader::new(file), path_buf))
     }
+
+    fn get_rc_bytes(&self, req: &DataRequest) -> Result<(Rc<[u8]>, PathBuf), DataError> {
+        let (mut reader, path_buf) = self.get_reader(req)?;
+        let mut buffer = Vec::<u8>::new();
+        reader
+            .read_to_end(&mut buffer)
+            .map_err(|e| DataError::Resource(Box::new(Error::Io(e, Some(path_buf.clone())))))?;
+        let rc_bytes: Rc<[u8]> = buffer.into();
+        Ok((rc_bytes, path_buf))
+    }
 }
 
 impl<'d, 's, M> DataProvider<'d, 's, M> for FsDataProvider
 where
     M: DataMarker<'s>,
-    // TODO(#667): Change this to Deserialize<'s>
-    M::Yokeable: serde::de::DeserializeOwned,
+    for<'de> SerdeDeDataStructWrap<<M::Yokeable as icu_provider::yoke::Yokeable<'de>>::Output>:
+        serde::de::Deserialize<'de>,
```

Comment:
Any suggestions on how to make this type bound prettier, or should I leave it like this?

Reply:
Well, I would import `Yokeable` and `Deserialize`, really shouldn't be using full paths in bounds

I'd also add a comment with the non-hacky bound, like `// for<'de> <M::Yokeable as Yokeable<'de>>::Output: Deserialize<'de>`, as I mentioned above

Reply:
I imported `Yokeable`.

I don't like importing `Deserialize` because there is also `erased_serde::Deserialize` used in the same file, and it is confusing to mix them up.  I prefer to fully qualify the two versions of the `Deserialize` trait.